### ISSUE TITLE
TypeError: attachments.forEach is not a function (BotController.js:136)

### ISSUE THREAD

**Issue by fardin01** [Freeways/messenger-bot-sails#2]
When sending a picture to the bot, the below error shows up:

`TypeError: attachments.forEach is not a function (BotController.js:136)
`
for the below code:

```
var handleAttachments = function (user, attachments) {
  attachments.forEach(function (attachment) {
       ...
    })
}
```

**Comment by marrouchi** [comment]
Hi,

handleAttachments() is a function that handles the event when the user sends an attachment to the chatbot (a picture for example) and not the other way around.

In the other hand it looks like when you did call this function, you did not add the "attachments" argument.

**Comment by fardin01** [comment]
Thanks @marrouchi  , this got clear.

But now I'm running into this error:

```
TypeError: Cannot read property 'forEach' of undefined
    at /...../api/controllers/BotController.js:46:24
    at Array.forEach (<anonymous>)

```

For the very beginning of the code:



```
    data.entry.forEach(function (entry) {
    // Iterate over each messaging event
    entry.messaging.forEach(function (messaging) {
   ....
   }

```
The first `forEach `works perfectly, but the second `forEach `is recognized as a property of `messaging`!

Is this a bug?

**Comment by marrouchi** [comment]
Y're welcome !

I think it would be better to test if entry does contain messaging as an array property, before doing any forEach.

Could you make a pull request to fix this bug ?

**Comment by fardin01** [comment]
@marrouchi, the first issue that I added here is because in `message ` model attachment is type string (so "[object Object]" is stored in MobgoDB), but later on `attachments.forEach()` is called on it, hence `TypeError: attachments.forEach is not a function`

**Comment by bashz** [comment]
This was just fixed on  6ed9c8b
